http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java

```diff
@@ -212,7 +214,8 @@ private SdkEventLoopGroup nonManagedEventLoopGroup(SdkEventLoopGroup eventLoopGr
     @Override
     public void close() {
         runAndLogError(log, "Unable to close channel pools", pools::close);
-        runAndLogError(log, "Unable to shutdown event loop", sdkEventLoopGroup.eventLoopGroup()::shutdownGracefully);
+        Future<?> closePromise = sdkEventLoopGroup.eventLoopGroup().shutdownGracefully();
+        runAndLogError(log, "Unable to shutdown event loop", () -> syncInterruptibly(closePromise));
```

Comment:
Should we do the acquiring of the closePromise within the runAndLogError as well so that close() won't throw any exceptions?

Reply:
+1